autorest/codegen/models/credential_model.py

```diff
@@ -0,0 +1,55 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import List, Optional, Type
+from .credential_schema_policy import CredentialSchemaPolicy, BearerTokenCredentialPolicy
+from .credential_schema_policy import ARMChallengeAuthenticationPolicy
+from .credential_schema import TokenCredentialSchema, AzureKeyCredentialSchema
+
+
+class CredentialModel:
+    """Store info about credential.
+    """
+
+    def __init__(
+        self,
+        azure_arm: bool
+    ) -> None:
+        self.azure_arm: bool = azure_arm
+        self.credential_scopes: List[str] = []
+        self.key_header_name: str = ""
+        self.policy_type: Optional[Type[CredentialSchemaPolicy]] = None
+        self._credential_schema_policy: Optional[CredentialSchemaPolicy] = None
+
+    @staticmethod
+    def aad_type() -> str:
```

Comment:
overall looks really good, only thing is I don't think we need these two static methods as part of the credential model. They're security specific, and we want the credential model to not know anything about "security" or "credential flags". Also, this code is only used in codegen/__init__.py so there's no need to have it here as well